addons/im_livechat/static/tests/embed/feedback_panel.test.js

```diff
@@ -51,6 +51,8 @@ test("Close without feedback", async () => {
     triggerHotkey("Enter");
     await contains(".o-mail-Message-content", { text: "Hello World!" });
     await click("[title*='Close Chat Window']");
+    await contains(".o-livechat-closeConfirmation-overlay");
+    await click(".o-livechat-closeConfirmation-leave");
     await click("button", { text: "Close conversation" });
     await contains(".o-livechat-LivechatButton");
```

Comment:
You should make only 1 test with the `contains(".o-livechat-CloseConfirmation-overlay")`, the others make tests more noisy than it should.

IMO I see no point in testing this classname. It is pure invented semantics that could mean anything (I could forge this classname somewhere and the test passes, it doesn't say much). In general we prefer `contains()` on something relevant to end-user, e.g. `text: {}` or `[title='']`. When we target a specific class, usually this means a big component that is already tests enough so checking classname is good enough to confidently say all related features are available.

And yes, many of our tests are asserting presence of some very specific classnames. Yes are of poor quality 🥲